### DISCUSSION TITLE
how to use custom api at preload index.js

### DISCUSSION THREAD

**Discussion by bsneric** [alex8088/electron-vite#577] [Q&A]
I found that there is a empty method which is at preload/index.js as followL

```
// Custom APIs for renderer
const api = {
// Can I put some method on here to open a file dialog?
}



```

My question is : is there a demo for how to use it? for example, I have a button on App.vue which is for open a dialog to select the files on my OS. Can I write a method in this custom api . if it is , how to make it. Could you please give me an example ?

**Comment by alex8088** [comment]
https://electron-vite.org/guide/dev#using-preload-scripts
